src/main/java/seedu/address/logic/commands/DoneCommand.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.results.CommandResult;
+import seedu.address.model.Model;
+import seedu.address.model.item.Note;
+
+/**
+ * This command allows user to mark a note as done.
+ */
+public class DoneCommand extends Command {
+    public static final String COMMAND_WORD = "done";
+
+    public static final String MESSAGE_DONE_SUCCESS = "Mark this note as done!";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " INDEX \n"
+            + "Example: done 1 ";
+
+    protected final Index targetIndex;
+
+    public DoneCommand(Index targetIndex) {
+        requireNonNull(targetIndex);
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (targetIndex.getZeroBased() >= model.getNoteListSize()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+        }
+
+        Note toDone = model.getNote(targetIndex);
+        toDone.markAsDone();
+        model.setNote(toDone, toDone);
+        model.commitResumeBook();
+
+        return new CommandResult(toDone.toString(), MESSAGE_DONE_SUCCESS, "");
```

Comment:
We now have abstracted all `CommandResult` to have a specific subclass. To be consistent, can you create `DoneCommandResult` as well?